### ISSUE TITLE
Autoselect next worker too for kicking

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#6297]
[<img align='left' src='https://codeberg.org/avatars/593dc43e2444c25db9481960e5b693ab' alt='tothxa' height='48' />](https://codeberg.org/tothxa)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/4660)*
*Created on Sun Jan 07 11:18:38 CET 2024 by [**Tóth András (tothxa)**](https://codeberg.org/tothxa)*

---

**Is your feature request related to a problem? Please describe.**
Re https://codeberg.org/wl/widelands/pulls/4657#issuecomment-1422937 / continuation of https://github.com/widelands/widelands/pull/5515 and #6196

**Describe the solution you'd like**
When no worker is selected for kicking, pressing the kick worker button should evict the next one when the first slot is empty, ie. it should be possible to kick all workers one-by-one, by clicking the button repeatedly. (maybe even kick all at once with ctrl+clicking the button?)

If a slot is manually selected (has focus), then the button should probably only affect the worker in that one slot. (except ctrl+click / evict all, which should work in this case too)

**Comment by bunnybot** [comment]
<img align='left' src='https://codeberg.org/avatars/593dc43e2444c25db9481960e5b693ab' alt='tothxa' height='48' />*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/4660#issuecomment-1423363)*
*On Sun Jan 07 11:32:25 CET 2024, **Tóth András (tothxa)** wrote:*

---

> If a slot is manually selected (has focus)...

Oops, looks like this is not so easy, the button gets the focus on click, so there's no difference if the worker was manually selected or autoselected. OK, then the next one can be autoselected too, when the first selection was manual.
